libtransmission/blocklist.cc

```diff
@@ -576,14 +576,21 @@ void BlocklistFile::assertValidRules(std::vector<AddressRange> const& ranges)
 
     for (size_t i = 1; i < std::size(ranges_ipv4); ++i)
     {
-        TR_ASSERT(ranges_ipv4[i - 1].end_ < ranges_ipv4[i].begin_);
+        if (ranges_ipv4[i - 1].end_ < ranges_ipv4[i].begin_)
```

Comment:
This `if` is simply silencing the assertion below, since the condition is identical to the assertion. Your code is equivalent to `/* do nothing */`.